industry_benchmarks/utils/results_cleanup.py

```diff
@@ -0,0 +1,349 @@
+#!/usr/bin/env python
+
+import argparse
+import json
+import os
+import pathlib
+import traceback
+from pathlib import Path
+from shutil import copyfile
+
+import MDAnalysis as mda
+import numpy as np
+import yaml
+from openfe_analysis import FEReader
+from openmmtools import multistate
+from tqdm import tqdm
+
+# TODO test gather before and after
+# TODO add more print statements
+# TODO check size diff in removing PDB + ligands
+
+
+def remove_first_reversed_sequential_duplicate_from_path(path: Path) -> Path:
+    """
+    Remove the first duplicated directory from path
+    We reverse the path so we remove the first duplicated directory
+    starting from the deepest part of the path
+    """
+
+    # reverse the path parts so we start from the deepest part first
+    reversed_path_parts = list(reversed(path.parts))
+    max_idx = len(reversed_path_parts)
+
+    # find index of first dupe
+    for idx in range(max_idx - 1):
+        if reversed_path_parts[idx] == reversed_path_parts[idx + 1]:
+            break
+    # no dupes
+    else:
+        print("Path didn't have any dupes")
+        return path
+
+    del reversed_path_parts[idx]
+    return Path(*reversed(reversed_path_parts))
+
+
+def compute_mbar_energies(analyzer):
+    """
+    Returns
+    ------
+    u_ln: energy matrix of shape (K,N) indexed by k,n
+        K is the total number of states observables are desired.
+        N is the total number of samples drawn from ALL states.
+        The nth configuration is the energy evaluated in the kth thermodynamic state.
+    N_l: 1-D iterable of shape K
+        The number of samples drawn from each kth state.
+    """
+    analyzer.use_full_trajectory = True
+    u_ln, N_l = analyzer._compute_mbar_decorrelated_energies()
+    return u_ln, N_l
+
+
+def get_replica_state_indices(analyzer):
+    energy_data = list(analyzer._read_energies(truncate_max_n_iterations=True))
+    replicas_state_indices = energy_data[-1]
+    return replicas_state_indices
+
+
+def subsample_traj(simulation, hybrid_system_pdb, lambda_windows, outfile):
+    for i in range(0, lambda_windows):
+        u = mda.Universe(hybrid_system_pdb, simulation, format=FEReader, state_id=i)
+        frames = [round(i) for i in np.linspace(0, len(u.trajectory) - 1, 21)]
+        out_traj = pathlib.Path(f"{outfile}_{i}.xtc")
+        with mda.Writer(str(out_traj), n_atoms=len(u.atoms)) as w:
+            for ts in u.trajectory[frames]:
+                w.write(u.atoms)
+
+
+def extract_data(simulation, checkpoint, hybrid_pdb, outfile, out_traj="out"):
+    """
+    Extract the MBAR-ready energy matrix and replica state indices.
+
+    Also extracts some info about the simulation and saves info in a
+    info.yaml file in the same folder as out_traj
+
+    Parameters
+    ----------
+    simulation: pathlib.Path
+        Path to the simulation `.nc` file
+    checkpoint: pathlib.Path
+        Path to the checkpoint `.chk` file
```

Comment:
TODO: This shouldn't be a path but the name of the file.

Reply:
I don't think it is really used but we can fix it (the code works and I pass it the path :shrug: )

Reply:
I did change it to work with just the file name.